src/snowflake/snowpark/modin/pandas/base.py

```diff
@@ -3892,6 +3892,10 @@ def __getitem__(self, key):
         if isinstance(self, pd.Series):
             return self.loc[key]
 
+        # Sometimes the result of a callable is a DataFrame (e.g. df[df > 0]) - use where.
```

Comment:
Can we add a TODO to upstream this work cc @sfc-gh-joshi @sfc-gh-mvashishtha

Reply:
This already works in upstream Modin.
```
>>> import modin.pandas as pd
>>> data = {"a": [1, 2, 3], "b": [4, 5, 6]}
>>> df = pd.DataFrame(data)
>>> df[lambda x: x < 2]
     a   b
0  1.0 NaN
1  NaN NaN
2  NaN NaN
>>> df[df < 2]
     a   b
0  1.0 NaN
1  NaN NaN
2  NaN NaN
```